src/magicbox/SuggestionsManager.ts

```diff
@@ -267,38 +305,35 @@ export class SuggestionsManager {
   }
 
   private buildPreviewContainer() {
-    return $$('div', {
-      className: 'coveo-preview-container'
-    }).el;
-  }
-
-  private get querySuggestPreviewComponent() {
-    const querySuggestPreviewElement: HTMLElement = $$(this.root).find(`.${Component.computeCssClassNameForType('QuerySuggestPreview')}`);
-    if (!querySuggestPreviewElement) {
-      return;
-    }
-    return Component.get(querySuggestPreviewElement);
+    return (this.previewContainer = $$(
+      'div',
+      {
+        className: 'coveo-preview-container'
+      },
+      $$('h4', {
+        className: 'coveo-preview-header'
+      }),
+      $$('div', {
+        className: 'coveo-preview-results'
+      })
+    )).el;
   }
 
-  private initPreviewForSuggestions(suggestions: Dom) {
-    const querySuggestPreview = this.querySuggestPreviewComponent;
-    if (!querySuggestPreview) {
-      return suggestions;
-    }
-
-    const suggestionContainerParent = $$('div', {
-      className: 'coveo-suggestion-container'
-    });
-
-    const previewContainer = this.buildPreviewContainer();
-    suggestionContainerParent.append(suggestions.el);
-    suggestionContainerParent.append(previewContainer);
-    return suggestionContainerParent;
+  private initPreviewForSuggestions() {
+    this.suggestionsPreviewContainer = $$(
+      'div',
+      {
+        className: 'coveo-suggestion-container'
+      },
+      this.suggestionsListbox.el,
+      this.buildPreviewContainer()
+    );
+    this.element.appendChild(this.suggestionsPreviewContainer.el);
   }
```

Comment:
This function wraps the already-appended suggestions list container in a new container along with the search result previews. The $$() function uses `appendChild` on its children, meaning that `this.suggestionsListbox.el` will not be duplicated, but rather moved to this new container.